esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -191,6 +191,82 @@ void CWsESDLConfigEx::init(IPropertyTree *cfg, const char *process, const char *
     m_esdlStore.setown(createEsdlCentralStore());
 }
 
+// Build two independent lists. First an array of IEspMethodConfig elements that lists
+// all methods defined across all services in the definition. Second is a StringArray
+// listing all the services defined in the definition.
+void CWsESDLConfigEx::buildServiceMethodsResponse(IEsdlDefinitionInfo* defInfo, IArrayOf<IEspMethodConfig>& methodList, StringArray& serviceList, const char* svc)
+{
+    StringArray* services = defInfo->queryServices();
+    ForEachItemIn(i, *services)
+    {
+        const char* serviceName = services->item(i);
+
+        // If a service name is passed in, then only return info
+        // for that single service
+        if( isEmptyString(svc) || strcasecmp(svc, serviceName)==0 )
+        {
+            serviceList.append(serviceName);
+            StringArray* methods = defInfo->queryMethods(serviceName);
+
+            ForEachItemIn(j, *methods)
+            {
+                Owned<IEspMethodConfig> methodconfig = createMethodConfig("","");
+                methodconfig->setName(methods->item(j));
+                methodList.append(*methodconfig.getClear());
+            }
+        }
+    }
+}
+
+// Builds an array of IEspESDLService elements, one for earch service in the
+// definition. Each element contains an array of of IEspMethodConfig elements,
+// one for each method defined in that service.
+void CWsESDLConfigEx::buildServiceWithMethodsResponse(IEsdlDefinitionInfo* defInfo, IArrayOf<IEspESDLService>& serviceList, const char* svc)
+{
+    StringArray* services = defInfo->queryServices();
+    ForEachItemIn(i, *services)
+    {
+        const char* serviceName = services->item(i);
+
+        // If a service name is passed in, then only return info
+        // for that single service
+        if( isEmptyString(svc) || strcasecmp(svc, serviceName)==0 )
+        {
+            Owned<IEspESDLService> esdlservice = createESDLService("","");
```

Comment:
Btw, I think in cases like this ("","") may be defaulted.  You may be able to use ();

Reply:
fixed

Reply:
changed